docker/lwua-dereferencer/dereferencer-py/dereferencer/dereference.py

```diff
@@ -1,128 +1,70 @@
 
-            if derefTask.file_name not in tasks_run:
-                log.info(f"task {derefTask.file_name} not in tasks_run")
-                derefTask.run_deref_task()
-                continue
+        log.info(TARGETSTORE)
 
-            # check if the task is expired
-            if isExpired(tasks_run[derefTask.file_name], derefTask.cache_lifetime):
-                derefTask.run_deref_task()
-                continue
+        CONFIGBUILDER = TravHarvConfigBuilder.TravHarvConfigBuilder(
+            TARGETSTORE,
+            str(config_folder_path),
+        )
 
-            log.info(
-                f"task {derefTask.file_name} was present in GraphDB and is not expired"
-            )
+        CONFIGLIST = CONFIGBUILDER.build_from_folder()
 
+        for travHarvConfig in CONFIGLIST:
+            log.info("Config object: {}".format(travHarvConfig()))
+            prefix_set = travHarvConfig.PrefixSet
+            config_name = travHarvConfig.ConfigName
+            tasks = travHarvConfig.tasks
 
-class DerefTask:
-    def __init__(self, config_file: Path):
-        self.store = Graph()
-        self.load_config(config_file)
-
-    def load_config(self, config_file):
-        with open(config_file, "r") as stream:
-            try:
-                config = yaml.safe_load(stream)
-                self.subjects = config["subjects"]
-                # check if child of subjects is SPARQL or literal
-                # if SPARQL then get the uri's from the SPARQL query
-                # if literal then get the uri's from the literal
-                if "SPARQL" in self.subjects:
-                    self.uris = uri_list(self.subjects["SPARQL"])
-                elif "literal" in self.subjects:
-                    self.uris = self.subjects["literal"]
-                else:
-                    log.error("subjects should contain either SPARQL or literal")
-                    raise Exception("subjects should contain either SPARQL or literal")
-                self.prefixes = {}
-                if "prefixes" in config:
-                    self.prefixes = config["prefixes"]
-
-                self.deref_paths = config["assert-paths"]
-                log.info(f"deref_paths: {self.deref_paths}")
-                self.cache_lifetime = (
-                    config["cache_lifetime"] if "cache_lifetime" in config else 0
-                )
-                self.file_name = config_file.name
-            except yaml.YAMLError as exc:
-                log.error(exc)
-
-    def write_store(self, filename):
-        """
-        Write the store to the graph database
-        """
-        log.info("writing store to graph database")
-        writeStoreToGraphDB(self.store, filename)
-
-    def run_deref_task(self):
-        self.store = Graph()
-        for uri in self.uris:
-            derefEntity = DerefUriEntity(
-                uri, self.deref_paths, self.store, self.prefixes
+            travharvexecutor = TravHarvExecuter.TravHarvExecutor(
```

Comment:
trav_harv_executor

Reply:
the double uppercase on  the right seems odd --> I am afraid it is a consequence of capitalisation in python source file `TravHarvExceuter.py` ?  

point being --> uppercase indicate class for me -- in this case that would be a nested class?

(so unsure what flake or other recommendations around that exist? or if we should not urgently make those part of our own conventions?)

Reply:
Will be resolved in the pyTravHarv module itself , the api-implementation that you see here is subject to change still.